unix_integration/pam_kanidm/src/pam/mod.rs

```diff
@@ -203,14 +203,14 @@ impl PamHooks for PamKanidm {
             Err(e) => return e,
         };
 
-        let mut daemon_client =
-            match DaemonClientBlocking::new(cfg.sock_path.as_str(), cfg.unix_sock_timeout) {
-                Ok(dc) => dc,
-                Err(e) => {
-                    error!(err = ?e, "Error DaemonClientBlocking::new()");
-                    return PamResultCode::PAM_SERVICE_ERR;
-                }
-            };
+        let mut timeout = cfg.unix_sock_timeout;
```

Comment:
I had to add `call_and_wait_timeout` because of the new DaemonClientBlocking struct, which stores the timeout. This might have seemed reasonable at the time, but we need to be able to modify the timeout to do a DeviceAuthorizationGrant (because the daemon may take up to `DeviceAuthorizationResponse.expires_in` seconds to respond (which defaults to 900 sec/15 min).

Reply:
This workaround fails, and it still times out after the default timeout. If I set the config timeout to something larger, than that works. The `DaemonClientBlocking` is ignoring my attempt to change the timeout.

Reply:
Resolved with the latest push.